### PR TITLE
E1601 Hyperlinks methods

### PR DESCRIPTION
**Pull request by hkumbum**
1. Deleted the hyperlink related methods from Assignment participant. 
2. Created submitted_hyperlinks, hyperlinks, remove_hyperlink methods in Assignment Team. (hyperlinks_array was not created. Only 'hyperlinks' method would be used by all callers). Also, has_submissions already existed in Assignment Team so we just refactored it.
3. All callers of earlier Assignment Participants methods were refactored to use the new methods in Assignment Team. We got Assignment Participant's Assignment team using 'team' method of Assignment Participants.
4. Corresponding controller methods for these model methods were also refactored.

### CODE REVIEW

File: spec/models/assignment_team_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe AssignmentTeam, type: :model do
+  # pending "add some examples to (or delete) #{__FILE__}"
+
+  let(:team){AssignmentTeam.new name: "Wikipedia contribution_Team2", parent_id: 754, type: "AssignmentTeam", submitted_hyperlinks: "---\n- http://water.com\n- http://shed.com" }
```
Comment (yangsong8) [inline spec/models/assignment_team_spec.rb]: Hi guys, please read bullet points 5 under "On all refactoring" in your OSS project write-up. In the testing db, there is only very limited data. You cannot assume assignment 754 is in the testing db.

**Comment by yangsong8** [comment]
Hi team, could you synchronize the recent changes of the code on master branch? After this, Travis CI would be able to check your pull request.

File: app/models/student_task.rb [hunk 2]
```diff
@@ -48,7 +48,7 @@ def course
   end
 
   def hyperlinks
-    @hyperlinks ||= participant.hyperlinks
+    @hyperlinks ||= participant.team.nil? ? [] : participant.team.hyperlinks
```
Comment (yangsong8) [inline app/models/student_task.rb]: Is this code ever called anywhere?

Comment (aneesh87) [inline app/models/student_task.rb]: Yes, In the same model app/models/student_task.rb, method name content_submitted_in_current_stage?, it calls hyperlinks.present? in checking if content has been submitted.

**Comment by yangsong8** [comment]
Hi guys, thank you for your work. We will merge your project with minor changes (on the tests). We appreciate your contribution to Expertiza!

**Pull request merged**